framework/test/unit/modules/interoperability/mainchain/commands/sidechain_registration.spec.ts

```diff
@@ -0,0 +1,534 @@
+import {
+	CCM_STATUS_OK,
+	COMMAND_ID_SIDECHAIN_REG,
+	CROSS_CHAIN_COMMAND_ID_REGISTRATION,
+	EMPTY_FEE_ADDRESS,
+	EMPTY_HASH,
+	MAINCHAIN_ID,
+	MAX_UINT64,
+	MODULE_ID_INTEROPERABILITY,
+	STORE_PREFIX_CHAIN_DATA,
+	STORE_PREFIX_CHAIN_VALIDATORS,
+	STORE_PREFIX_CHANNEL_DATA,
+	STORE_PREFIX_OUTBOX_ROOT,
+	STORE_PREFIX_REGISTERED_NAMES,
+	STORE_PREFIX_REGISTERED_NETWORK_IDS,
+} from '../../../../../../src/modules/interoperability/constants';
+import {
+	nameSchema,
+	chainIDSchema,
+	sidechainRegParams,
+	chainAccountSchema,
+	channelSchema,
+	validatorsSchema,
+	outboxRootSchema,
+	registrationCCMParamsSchema,
+} from '../../../../../../src/modules/interoperability/schema';
+import { SidechainRegistrationParams } from '../../../../../../src/modules/interoperability/types';
+import { CommandVerifyContext, VerifyStatus } from '../../../../../../src/node/state_machine';
+import { computeValidatorsHash } from '../../../../../../src/modules/interoperability/utils';
+
+describe('Sidechain registration command', () => {
+	const transactionParams = {
+		name: 'sidechain',
+		genesisBlockID: Buffer.alloc(0),
+		initValidators: [
+			{
+				blsKey: Buffer.from(
+					'3c1e6f29e3434f816cd6697e56cc54bc8d80927bf65a1361b383aa338cd3f63cbf82ce801b752cb32f8ecb3f8cc16835',
+					'hex',
+				),
+				bftWeight: BigInt(10),
+			},
+			{
+				blsKey: Buffer.from(
+					'4c1e6f29e3434f816cd6697e56cc54bc8d80927bf65a1361b383aa338cd3f63cbf82ce801b752cb32f8ecb3f8cc16835',
+					'hex',
+				),
+				bftWeight: BigInt(10),
+			},
+		],
+		certificateThreshold: BigInt(10),
+	};
+	const encodedTransactionParams = codec.encode(sidechainRegParams, transactionParams);
+	const publicKey = getRandomBytes(32);
+	const transaction = new Transaction({
+		moduleID: MODULE_ID_INTEROPERABILITY,
+		commandID: COMMAND_ID_SIDECHAIN_REG,
+		senderPublicKey: publicKey,
+		nonce: BigInt(0),
+		fee: BigInt(100000000),
+		params: encodedTransactionParams,
+		signatures: [publicKey],
+	});
+	const networkIdentifier = Buffer.from(
+		'e48feb88db5b5cf5ad71d93cdcd1d879b6d5ed187a36b0002cc34e0ef9883255',
+		'hex',
+	);
+	const networkID = hash(Buffer.concat([Buffer.alloc(0), transaction.senderAddress]));
+	let sidechainRegistrationCommand: SidechainRegistrationCommand;
+	let db: KVStore;
+	let stateStore: StateStore;
+	let nameSubstore: StateStore;
+	let networkIDSubstore: StateStore;
+	let verifyContext: CommandVerifyContext<SidechainRegistrationParams>;
+
+	beforeEach(() => {
+		sidechainRegistrationCommand = new SidechainRegistrationCommand(
+			MODULE_ID_INTEROPERABILITY,
+			new Map(),
+			new Map(),
+		);
+		db = new InMemoryKVStore() as never;
+		stateStore = new StateStore(db);
+		nameSubstore = stateStore.getStore(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_REGISTERED_NAMES);
+		networkIDSubstore = stateStore.getStore(
+			MODULE_ID_INTEROPERABILITY,
+			STORE_PREFIX_REGISTERED_NETWORK_IDS,
+		);
+	});
+
+	describe('verify', () => {
```

Comment:
I think it'd be nice to also have tests for :
1. `trs.params.name` should be at most MAX_LENGTH_NAME characters long
2. `trs.params.initValidators` array must have at least 1 element and at most MAX_NUM_VALIDATORS elements.
3. The `blsKey` property of each element has a length of 48 bytes

as mentioned in the LIP for the [verify](https://github.com/LiskHQ/lips/blob/aced79e3a90c893f25c8f0735ce73732f17cced6/proposals/lip-0043.md#verification) hook.